src/test/java/io/strimzi/kafka/metrics/KafkaPrometheusMetricsReporterTest.java

```diff
@@ -38,31 +38,36 @@ public void testLifeCycle() throws Exception {
         KafkaPrometheusMetricsReporter reporter = new KafkaPrometheusMetricsReporter(new PrometheusRegistry());
         Map<String, String> configs = new HashMap<>();
         configs.put(PrometheusMetricsReporterConfig.LISTENER_CONFIG, "http://:0");
+        configs.put(PrometheusMetricsReporterConfig.ALLOWLIST_CONFIG, "kafka_server_group_name.*");
         reporter.configure(configs);
         reporter.contextChange(new KafkaMetricsContext("kafka.server"));
+
         Optional<Integer> port = reporter.getPort();
         assertTrue(port.isPresent());
         int initialMetrics = getMetrics(port.get()).size();
 
-        KafkaMetric metric1 = buildMetric("name1", "group", 0);
+        // Adding a metric not matching the allowlist does nothing
+        KafkaMetric metric1 = buildMetric("other", "group", 0);
         reporter.init(Collections.singletonList(metric1));
-
         List<String> metrics = getMetrics(port.get());
-        assertEquals(initialMetrics + 1, metrics.size());
+        assertEquals(initialMetrics, metrics.size());
 
-        KafkaMetric metric2 = buildMetric("name2", "group", 0);
+        // Adding a metric not matching the allowlist does nothing
```

Comment:
This comment does not look right, this metric does match the allowlist

Reply:
Fixed